server/src/test/java/io/crate/execution/dml/IndexerTest.java

```diff
@@ -815,6 +816,30 @@ public void test_array_length_is_indexed() throws Exception {
         assertTranslogParses(doc, table);
     }
 
+    @Test
+    public void test_array_length_is_indexed_for_child_arrays() throws Exception {
```

Comment:
I do not see how we can benefit from these `_array_length_` indexes since each element of the array are indexed with the same name, i.e. array `o = [{c=[1,2]}, {c=[1]}]` indexed as `IntField(_array_length_**c**, 2)`,`IntField(_array_length_**c**, 1)` **not** `IntField(_array_length_**c1**, 2)`,`IntField(_array_length_**c2**, 1)`.

If these `_array_length_` indexes are to be removed for space efficiency, we should think about indexes of array of array altogether. I think currently we fall back to generic function queries immiediately for any queries involving array elements resulting in subscript functions.

Reply:
Getting back to https://github.com/crate/crate/pull/16436#discussion_r1699458548

>is that true that array sub-cols are not indexed

Asking about code comment (fell like it has to be rephrased):
so as you said, we have 2 _array_length_X fields. 
New `IndexerTest` shows that they are indexed?

>If these _array_length_ indexes are to be removed
This is not (yet) done and thus code comment is not correct?

Sorry, it's actually a minor comment despite in being verbose, I just got confused a bit.

Reply:
Thanks for reviewing, I will surely re-phrase this.

> but `o_array['xs']` is not indexed.

I meant to say that `o_array['xs']` as a read reference is not indexed. Lets say we indexed `o_array = [{xs=[1,2]}, {XS=[1]}]`. The read reference of `o_array['xs']` is an `int_array_array`, `[[1,2], [1]]`. `_array_length_` for such nested array was never indexed but should be equal to  `_array_length_o_array`.

> New IndexerTest shows that they are indexed?
> This is not (yet) done and thus code comment is not correct?

Yes they are indexed and here I meant `o_array['xs']` as a `int_array`, `[1,2]` and `[1]`.

Reply:
>o_array['xs'] as a read reference is not indexed

I checked again existing code and how we build read refs, it makes sense, now, thanks.